code/modules/mob/living/simple_animal/constructs/constructs.dm

```diff
@@ -233,6 +237,7 @@
 	resistance = 10
 	var/energy = 0
 	var/max_energy = 1000
+	can_climb = 1 //It flies, its magic
```

Comment:
juggernauts are capable of manufacturing enough momentum to smash themselves through walls, using only magic, they float and don't push off any other surface. It seems pretty reasonable that they could use the same ability to slowly climb up a ladder, especially when they can use their hands to grab on and pull up